buildkite/resource_cluster_queue.go

```diff
@@ -0,0 +1,242 @@
+		MarkdownDescription: "A Cluster Queue is a queue belonging to a specific Cluster for its Agents to target builds on. ",
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"cluster_uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"cluster_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The ID of the Cluster that this Cluster Queue belongs to.",
+			},
+			"key": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The key of the Cluster Queue.",
+			},
+			"description": resource_schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "A description for the Cluster Queue. ",
+			},
+		},
+	}
+}
+
+func (cq *ClusterQueueResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan, state ClusterQueueResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Creating cluster queue with key %s into cluster %s ...", plan.Key.ValueString(), plan.ClusterId.ValueString())
+	apiResponse, err := createClusterQueue(
+		cq.client.genqlient,
+		cq.client.organizationId,
+		plan.ClusterId.ValueString(),
+		plan.Key.ValueString(),
+		plan.Description.ValueStringPointer(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create Cluster Queue",
+			fmt.Sprintf("Unable to create Cluster Queue: %s", err.Error()),
+		)
+		return
+	}
+
+	state.Id = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Id)
+	state.Uuid = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Uuid)
+	state.ClusterId = plan.ClusterId
+	state.ClusterUuid = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Cluster.Uuid)
+	state.Key = types.StringValue(apiResponse.ClusterQueueCreate.ClusterQueue.Key)
+	state.Description = types.StringPointerValue(apiResponse.ClusterQueueCreate.ClusterQueue.Description)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (cq *ClusterQueueResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state ClusterQueueResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Getting cluster queues for cluster %s ...", state.ClusterUuid.ValueString())
+	queues, err := getClusterQueues(cq.client.genqlient, cq.client.organization, state.ClusterUuid.ValueString())
```

Comment:
I notice the `cluster` field on `Organization` has an `id` argument, but the docs for that say its a UUID which I imagine is why you are using `state.ClusterUuid` here. Is that correct? If it is, that seems like a bug in the GraphQL because everywhere else `id` is used as an argument, its the Relay Node ID, not the UUID.

It's probably not something we can really fix, but something to be aware of

Reply:
Correct - described [here](https://buildkite.com/user/graphql/documentation/type/Organization) - needing the `UUID` since the query is based from an `Organisation` query 👍